storage/shard.go

```diff
@@ -1542,6 +1571,68 @@ func (s *dbShard) Flush(
 	return s.markFlushStateSuccessOrError(blockStart, multiErr.FinalError())
 }
 
+func (s *dbShard) Snapshot(
+	blockStart time.Time,
+	snapshotStart time.Time,
+	flush persist.Flush,
+) error {
+	// We don't snapshot data when the shard is still bootstrapping
+	s.RLock()
+	if s.bs != bootstrapped {
+		s.RUnlock()
+		return errShardNotBootstrappedToSnapshot
+	}
+	s.RUnlock()
+
+	var multiErr xerrors.MultiError
+
+	s.markIsSnapshotting()
+	defer func() {
+		s.markDoneSnapshotting(multiErr.FinalError() == nil, snapshotStart)
```

Comment:
nit: `mutliErr.Empty()` instead of `multiErr.FinalError() == nil`

Reply:
oh nice